### PR TITLE
Tests

### PR DESCRIPTION
**Pull request by bytedissident**
Run BDD tests on example app, should cover all actions (except selectEmebeddedCollectionItem)
Added small unit tests on Performance beta and other applicable spots. 

- Updated readme
- Added UI test for each action in example app
- Added performance test beta (https://www.objc.io/issues/3-views/moving-pixels-onto-the-screen/ )
- re-organized Example App 
- improved resetUI to work with more situations.

### CODE REVIEW

File: Behave/Source/Behave+ViewQueries.swift [hunk 1]
```diff
@@ -186,4 +186,14 @@ public extension Behaviour {
             }
         }
     }
+    
+    func wait(for cell: IndexPath, parent: UITableView, complete: @escaping (UITableViewCell?) -> Void) {
```
Comment (denpef) [inline Behave/Source/Behave+ViewQueries.swift:190]: I doubt between the conciseness of writing and the detailing of the func parameters like `parentTable`, or even `waitForCell(at indexPath: IndexPath, tableView: UITableView`
To be honest, I first looked at the tests and did not understand what was happening and only then looked at the implementation 🙂

File: Behave/Source/Behave+Performance.swift [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    func setUpPerformanceTest() {
+        reset()
+        start = NSDate().timeIntervalSince1970
+    }
+    
+    /// Measures performance based on time in relation for FPS. Given that one frame is displayed in 16.67ms at 60 fps. Then the Number of frames * 16.67 shouldl be the total time it takes to display the frames in question.
+    /// - Parameters:
+    ///   - frames: the number of frames displayed in this view
+    /// - Returns: Bool
+    func measurePerformance(frames: Int) -> Bool {
```
[the remark this answers is not in the document]
Comment (bytedissident) [reply Behave/Source/Behave+Performance.swift:25]: @denpef by API example you mean the iOS 14 UI tests (https://developer.apple.com/videos/play/wwdc2020/10077/)? What I was trying to achieve is doing it w/out UI tests as this requires a real device, as well as things like coverage being turned off.

File: Behave/Source/Behave+ViewQueries.swift (hunk 1 above)
Comment (bytedissident) [reply Behave/Source/Behave+ViewQueries.swift:190]: @denpef so what is the suggestion here?

File: Behave/Source/Behave+Performance.swift (hunk 2 above)
[the remark this answers is not in the document]
Comment (bytedissident) [reply Behave/Source/Behave+Performance.swift]: @denpef @Vadimkomis I re-wrote the performance test api, to be accurate. Requesting new review.

File: BehaveTests/Behave+PerformanceTests.swift [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        // SET SOME BASE VALUES
+        sut.frameStart = 0.0167
+        sut.frameStart = 0.0334
+        sut.passesPerformanceTest = true
+        
+        sut.calculatePerformance()
+        XCTAssert(sut.passesPerformanceTest)
+    }
+    
+    func testCalculatePerformance_false() {
```
Comment (Vadimkomis) [inline BehaveTests/Behave+PerformanceTests.swift:47]: This test is failing is that on purpose ?

**Review by Vadimkomis — COMMENTED** [review]
@bytedissident I pushed a minor fix for non compiling  tests in order to fix them. I hope that's cool

File: BehaveTests/Behave+PerformanceTests.swift (hunk 3 above)
Comment (Vadimkomis) [reply BehaveTests/Behave+PerformanceTests.swift:47]: Also, maybe it makes sense to add a link to this article in this pr's description: https://www.objc.io/issues/3-views/moving-pixels-onto-the-screen/ ?
I think it adds context to this pr.

Comment (bytedissident) [reply BehaveTests/Behave+PerformanceTests.swift:47]: I fixed that test I had both values set to frameStart, instead of frameStart and frameEnd

**Comment by bytedissident** [comment]
> @bytedissident I pushed a minor fix for non compiling tests in order to fix them. I hope that's cool

sure

**Review by denpef — COMMENTED** [review]
Looks good to me 👍 
@Vadimkomis PR is ready for you

**Review by Vadimkomis — APPROVED** [review]
Looks good to me 👍

**Pull request merged**